core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3059,10 +3060,27 @@ private boolean syncMetadata(RpcContext rpcContext, LockedInodePath inodePath,
         // The requested path does not exist in Alluxio, so just load metadata.
         pathsToLoad.add(inodePath.getUri().getPath());
       } else {
-        SyncResult result =
-            syncInodeMetadata(rpcContext, inodePath, syncDescendantType);
-        deletedInode = result.getDeletedInode();
-        pathsToLoad = result.getPathsToLoad();
+        AlluxioURI path = inodePath.getUri();
+        MountTable.Resolution resolution = mMountTable.resolve(path);
+        AlluxioURI ufsUri = resolution.getUri();
+        try (CloseableResource<UnderFileSystem> ufsResource = resolution.acquireUfsResource()) {
+          UnderFileSystem ufs = ufsResource.get();
+          ListOptions listOptions = ListOptions.defaults();
+          if (syncDescendantType == DescendantType.ALL) {
+            listOptions.setRecursive(true);
+          } else {
+            listOptions.setRecursive(false);
+          }
+          children = ufs.listStatus(ufsUri.toString(), listOptions);
+          for (UfsStatus childStatus : children) {
```

Comment:
Does `UfsStatus#getName` return the complete relative name from where we listed? (ie. would it be `a/b` or just `b`)?

Reply:
yes, it returns the complete relative name from where we listed.